### PR TITLE
Register providers at etos-api launch

### PR DESCRIPTION
**Pull request by andmat900**
### Applicable Issues

- [Register providers in etos-api](https://github.com/eiffel-community/etos/issues/217)

### Description of the Change

This change implements the registration of providers when etos-api is launched. 

The code is taken from etos-environment-provider. This is part of the bigger change of transforming the Environment Provider from a service to a library. 

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Andrei Matveyeu, andreimu@axis.com

### CODE REVIEW

File: python/src/etos_api/__init__.py [hunk 1]
```diff
@@ -56,3 +57,5 @@
     FastAPIInstrumentor().instrument_app(
         APP, tracer_provider=PROVIDER, excluded_urls="selftest/.*,logs/.*"
     )
+
+REG = RegisterProviders()
```
Comment (fredjn) [inline python/src/etos_api/__init__.py:61]: Why is this added here?

**Comment by fredjn** [comment]
Perhaps we should mention in the PR description that this is a move of functionality from the Environment Provider, and somehow reference the bigger change(s) to transform the Environment Provider form Service -> Library.

File: python/src/etos_api/__init__.py (hunk 1 above)
Comment (andmat900) [reply python/src/etos_api/__init__.py:61]: The initial idea was to place it in main.py. However this file seems to be the actual entry point that imports from main.py. 

Should this be moved to main.py or some other place?

**Comment by andmat900** [comment]
> Perhaps we should mention in the PR description that this is a move of functionality from the Environment Provider, and somehow reference the bigger change(s) to transform the Environment Provider form Service -> Library.

The PR description is now updated.

**Review by fredjn — APPROVED** [review]
